### ISSUE TITLE
Feature: Add support for recipes that expect ingredients to be passed from multiple sides

### ISSUE THREAD

**Issue by GreyMario** [CyclopsMC/IntegratedCrafting#16]
#### Issue type:

- :heavy_plus_sign: Feature request

____

#### Short description:

Take Mekanism's Metallurgical Infuser as an example: it asks for a primary input ingredient and an Infusion material, and it can't take both from the same side. Mekanism's Infusions are a little weird in that they store an internal buffer of material, but it's entirely possible to create recipes that specifically consume exactly the amount of infusion material supplied with no leftovers or deficit.

In short, the issue is that a machine may demand that resources be passed from multiple sides, but the Crafting Interface can only input a specific _type_ of resource to one targeted side.

(This also affects, for example, Applied Energistics 2's Inscriber, which violently demands distinct item input from _three_ sides to satisfy crafting requirements.)

**Comment by rubensworks** [comment]
Hmm, I see the use, but I'm giving this a lower priority, as this can currently already be achieved with a more elaborate setup involving a buffer chest.

**Comment by GreyMario** [comment]
I figured the chest solution would be the workaround hack for the time being (with the downside of only handling items).

Incidentally, can items, fluids, _and_ energy be supplied to a small crafting-type network though some wizardry with the crafting interface? Not that I can think of a good use for all three at once right at the moment, but I'm curious if any such interaction is available.

**Comment by rubensworks** [comment]
> Incidentally, can items, fluids, _and_ energy be supplied to a small crafting-type network though some wizardry with the crafting interface?

Untested: try letting the crafting interface point to an ID cable with a fluid, energy and item interface on different sides, and configure the crafting interface to forward the ingredient types to the proper side of the cable. This should forward all ingredient types to the subnetwork.

**Comment by GreyMario** [comment]
On the latest release `1.12.2-1.0.1`, that method worked exactly once (after fiddling with the system to actually get it to accept crafting the test recipe). It even correctly placed the ingredients into the interfaced containers, which were attached to the same interfaces that I was pushing to with the crafting interface. Afterward, it claimed that it was "not allowed".

_Maybe_ the latest patches to crafting have fixed that? Haven't installed the latest dev build.

**Comment by rubensworks** [comment]
> Afterward, it claimed that it was "not allowed".
> _Maybe_ the latest patches to crafting have fixed that? Haven't installed the latest dev build.

That sounds like a bug. Could you try it out and report it if you find anything?

**Comment by GreyMario** [comment]
For the sake of completeness, the initially observed behavior with the suggested build was indeed resolved with a newer version.
